tests/tests/state_rewind.rs

```diff
@@ -380,3 +402,238 @@ async fn backup_and_restore__should_work_with_state_rewind() -> anyhow::Result<(
     driver.kill().await;
     Ok(())
 }
+
+/// Get arbitrary coin from a wallet
+async fn get_wallet_coin(client: &FuelClient, wallet_address: &Address) -> Coin {
+    let coins = client
+        .coins(
+            wallet_address,
+            None,
+            PaginationRequest {
+                cursor: None,
+                results: 10,
+                direction: fuel_core_client::client::pagination::PageDirection::Forward,
+            },
+        )
+        .await
+        .expect("Unable to get coins")
+        .results;
+    coins
+        .into_iter()
+        .next()
+        .expect("Expected at least one coin")
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn dry_run__correct_utxoid_state_in_past_blocks() -> anyhow::Result<()> {
+    let mut rng = StdRng::seed_from_u64(1234);
+    let poa_secret = SecretKey::random(&mut rng);
+
+    let mut config = fuel_core::service::Config::local_node();
+    config.consensus_signer = SignMode::Key(Secret::new(poa_secret.into()));
+    config.utxo_validation = true;
+    let srv = FuelService::new_node(config).await.unwrap();
+    let client = FuelClient::from(srv.bound_address);
+
+    // First, distribute one test wallet to multiple addresses
+    let wallet_secret =
+        SecretKey::from_str(TESTNET_WALLET_SECRETS[1]).expect("Expected valid secret");
+    let wallet_address = Address::from(*wallet_secret.public_key().hash());
+    let coin = get_wallet_coin(&client, &wallet_address).await;
+
+    let wallets: Vec<SecretKey> = (0..2).map(|_| SecretKey::random(&mut rng)).collect();
+
+    let mut tx = TransactionBuilder::script(vec![], vec![]);
+    tx.max_fee_limit(0)
+        .script_gas_limit(1_000_000)
+        .with_gas_costs(GasCosts::free())
+        .add_unsigned_coin_input(
+            wallet_secret,
+            coin.utxo_id,
+            coin.amount,
+            coin.asset_id,
+            TxPointer::new(coin.block_created.into(), coin.tx_created_idx),
+        );
+    for key in wallets.iter() {
+        let public = key.public_key();
+        let owner: [u8; Address::LEN] = public.hash().into();
+
+        tx.add_output(Output::Coin {
+            to: owner.into(),
+            amount: coin.amount / (wallets.len() as u64),
+            asset_id: coin.asset_id,
+        });
+    }
+    let first_tx = tx.finalize_as_transaction();
+    let status = client.submit_and_await_commit(&first_tx).await.unwrap();
+    let ClientTransactionStatus::Success {
+        block_height: bh_first,
+        ..
+    } = status
+    else {
+        panic!("unexpected result {status:?}")
+    };
+
+    // Then, transfer one of these coins to anyther wallet
```

Comment:
A nit:

```suggestion
    // Then, transfer one of these coins to another wallet
```

Reply:
Not caught by the spell checker? Weird. I'll fix in #2491 as well.

Reply:
Fixed in 9eeee8412d.